File: VHBoomMenuButtonExample/DraggableVC.m
//
//  DraggableVC.m
//  VHBoomMenuButton
//
//  Created by viktorhuang on 2017/4/14.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

#import "DraggableVC.h"
#import "BuilderManager.h"
#import "VHBoomMenuButton.h"

@interface DraggableVC ()

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb;

@property (weak, nonatomic) IBOutlet UISwitch *draggableSwitch;
@property (weak, nonatomic) IBOutlet UISwitch *backgroundEffectSwitch;
@property (weak, nonatomic) IBOutlet UISwitch *shadowEffectSwitch;

@end

@implementation DraggableVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    UIEdgeInsets edgeInsets = self.bmb.edgeInsetsInSuperView;
    // Prevent the bmbs being dragged behind navigation bar.
    edgeInsets.top += 66;
    
    self.bmb.buttonEnum = VHButtonSimpleCircle;
    self.bmb.piecePlaceEnum = VHPiecePlace_DOT_9_1;
    self.bmb.buttonPlaceEnum = VHButtonPlace_SC_9_1;
    self.bmb.edgeInsetsInSuperView = edgeInsets;
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb.piecePlaceEnum]; i++)
    {
        [self.bmb addBuilder:[BuilderManager simpleCircleButtonBuilder]];
    }
}

- (IBAction)onSwitchValueChanged:(UISwitch *)sender
{
    if (sender == self.draggableSwitch)
    {
        self.bmb.draggable = sender.on;
    }
    else if (sender == self.backgroundEffectSwitch)
    {
        self.bmb.backgroundEffect = sender.on;
    }
    else if (sender == self.shadowEffectSwitch)
    {
        self.bmb.shadowEffect = sender.on;
    }
}

@end

File: VHBoomMenuButtonSwiftDemo/DraggableVC.swift
//
//  DraggableVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by viktorhuang on 2017/5/4.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class DraggableVC: BoomMenuButtonVC {

    @IBOutlet weak var bmb: BoomMenuButton!
    
    @IBOutlet weak var draggableSwitch: UISwitch!
    @IBOutlet weak var hasBackgroundSwitch: UISwitch!
    @IBOutlet weak var hasShadowSwitch: UISwitch!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Prevent the bmbs being dragged behind navigation bar.
        bmb.edgeInsetsInSuperView.top += 66
        
        bmb.buttonEnum = .simpleCircle
        bmb.piecePlaceEnum = .dot_9_1
        bmb.buttonPlaceEnum = .sc_9_1
        for _ in 0..<bmb.piecePlaceEnum.pieceNumber() {
            bmb.addBuilder(BuilderManager.simpleCircleButtonBuilder())
        }
    }

    @IBAction func onSwitchValueChanged(_ sender: UISwitch) {
        if sender == draggableSwitch {
            bmb.draggable = sender.isOn
        } else if sender == hasBackgroundSwitch {
            bmb.hasBackground = sender.isOn
        } else if sender == hasShadowSwitch {
            bmb.hasShadow = sender.isOn
        }
    }
}
